tests/spec/test_dagspec.py

```diff
@@ -1884,6 +1885,41 @@ def test_lazy_load_product_level_client(tmp_directory, tmp_imports,
     assert 'my_testing_module' in sys.modules
 
 
+@pytest.mark.parametrize('client_spec', [
+    'my_testing_module.get_db_client',
+])
+@pytest.mark.parametrize(
+    'tasks, expected',
+    [([{
+        'source': 'script.sql',
+        'product': 'another.csb',
+        'client': 'my_testing_module.get_db_client',
+        'product_client': 'my_testing_module.get_db_client'
+    }], "'.csb' is not a valid extension. Did you mean: .csv"),
+     ([{
+         'source': 'script.sql',
+         'product': 'another.parquets',
+         'client': 'my_testing_module.get_db_client',
+         'product_client': 'my_testing_module.get_db_client'
+     }], "'.parquets' is not a valid extension. Did you mean: .parquet")])
+def test_product_extension_typo(tasks, expected, tmp_directory, client_spec):
+    Path('script.sql').write_text("""
+    SELECT * FROM my_table
+    """)
+
+    with sqlite3.connect('my.db') as conn:
+        pd.DataFrame({'x': range(5)}).to_sql('my_table', conn)
+
+    data = {'tasks': tasks}
+
+    with pytest.raises(DAGSpecInitializationError) as excinfo:
+        spec = DAGSpec(data)
+        dag = spec.to_dag()
```

Comment:
I think `spec = DAGSpec(data)` is enough, that's where the exception is raised, I think you can remove the `to_dag()` and `build` and the test will still work

Reply:
So I checked that only DAGSpec(data) doesnt raise the original `ploomber.exceptions.SourceInitializationError`. The minimal set of statements that raises this error is 
```
spec = DAGSpec(data)
spec.to_dag()
```

However this testcase still passes (with to_dag() and build removed):
```
with pytest.raises(DAGSpecInitializationError) as excinfo:
     spec = DAGSpec(data)
```

If you think otherwise I'll remove the to_dag() as well.